tests/data_context/store/test_store_backends.py

```diff
@@ -369,11 +372,16 @@ def test_TupleGCSStoreBackend():
             "leakybucket", prefix="this_is_a_test_prefix"
         )
 
-        my_store.remove_key("leakybucket", prefix="this_is_a_test_prefix")
+        my_store.remove_key("leakybucket")
 
         from google.cloud.exceptions import NotFound
 
         try:
             mock_client.get_bucket.assert_called_once_with("leakybucket")
         except NotFound:
             pass
+
+    with patch("google.cloud.storage.Client", autospec=True) as mock_gcs_client:
```

Comment:
this is _very_  simple test that checks whether the `mock_gcs_client` object can throw an `InvalidKeyError`.  Potential to use Docker to create a "fake" GCS server would allow for the tests to be done for real

Reply:
:+1:

Another option would be to configure integration test infra that includes a real GCS setup. But that would be integration testing, not just unit testing.